src/main/java/it/gov/pagopa/payhub/activities/activity/utility/SendEmailIngestionFlowActivityImpl.java

```diff
@@ -0,0 +1,125 @@
+                    .orElseThrow(() -> new IngestionFlowNotFoundException("Cannot found ingestionFlow having id: "+ ingestionFlowFileId));
+            String ipaCode = ingestionFlowFileDTO.getOrg().getIpaCode();
+            UserInfoDTO userInfoDTO = userAuthorizationService.getUserInfo(ipaCode, ingestionFlowFileDTO.getOperatorName());
+            UserInfoDTO organizationUserInfo = userAuthorizationService.getUserInfo(ipaCode,ipaCode+WS_USER);
+            MailTo mailTo = getMailFromIngestionFlow(ingestionFlowFileDTO, success);
+            mailTo.setTo(new String[]{userInfoDTO.getEmail()});
+            if (organizationUserInfo!= null && StringUtils.isNotBlank(organizationUserInfo.getEmail()) &&
+                ! organizationUserInfo.getEmail().equalsIgnoreCase(userInfoDTO.getEmail())) {
+                    mailTo.setCc(new String[]{organizationUserInfo.getEmail()});
+            }
+            mailTo.setHtmlText(mailTo.getHtmlText());
+            sendMailService.sendMail(mailTo);
+        }
+        catch (Exception e){
+            log.error("Sending mail failed", e);
+            return false;
+        }
+        return true;
+    }
+
+    private MailTo getMailFromIngestionFlow(IngestionFlowFileDTO ingestionFlowFileDTO, boolean success) throws Exception {
+        String template = "";
+        String flowType = ingestionFlowFileDTO.getFlowFileType();
+        if (! flowType.equalsIgnoreCase("R")) {
+            log.error("Sending e-mail not supported for flow type: {}", flowType);
+            throw new IngestionFlowTypeNotSupportedException("Sending e-mail not supported for flow type: "+flowType);
+        }
+
+        String subject = success ? emailTemplatesConfiguration.getPaymentsReportingFlowOk().getSubject()
+                : emailTemplatesConfiguration.getPaymentsReportingFlowKo().getSubject();
+        String body = success ? emailTemplatesConfiguration.getPaymentsReportingFlowOk().getBody()
+                : emailTemplatesConfiguration.getPaymentsReportingFlowKo().getBody() ;
+
+        Map<String, String> mailMap = new HashMap<>();
+        mailMap.put(Constants.ACTUAL_DATE, MAIL_DATE_FORMAT.format(LocalDateTime.now()));
+        mailMap.put(Constants.FILE_NAME, ingestionFlowFileDTO.getFileName());
+        mailMap.put(Constants.TOTAL_ROWS_NUMBER, String.valueOf(ingestionFlowFileDTO.getNumTotalRows()));
+        mailMap.put(Constants.MAIL_TEXT, StringSubstitutor.replace(template, mailMap, "{", "}"));
+
+        MailTo mailTo = new MailTo();
+        mailTo.setMailSubject(StringSubstitutor.replace(subject, mailTo.getParams(), "{", "}"));
+        String htmlText = StringSubstitutor.replace(body, mailTo.getParams(), "{", "}");
+        String plainText = Jsoup.clean(htmlText, "", Safelist.none(), new Document.OutputSettings().prettyPrint(false));
+        mailTo.setHtmlText(plainText);
+        mailTo.setTemplateName(template);
```

Comment:
what is the purpose of this templateName variable?
```suggestion
        mailTo.setTemplateName(template);
```

Reply:
no more used: removed